### PR TITLE
Fix Issue #25137

### PR DESCRIPTION
**Pull request by jdjohnston**
- Invert boolean value returned by Windows FreeLibrary so it matches
  that of POSIX dlclose
- Add test of jl_dlclose/Libdl.dlclose to test/libdl.jl

### CODE REVIEW

**Comment by staticfloat** [comment]
This looks great to me, thanks for your contribution @jdjohnston!

**Review by staticfloat — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
libdl test failure on Win32 looks related; Win64 passed.

File: test/libdl.jl [hunk 1]
```diff
@@ -183,6 +183,20 @@ let dl = C_NULL
...
+# dlclose should return true on success and false on failure
+let dl = C_NULL
+    @test -1 == ccall(:jl_dlclose, Cint, (Ptr{Void},), dl)
+    @test !Libdl.dlclose(dl)
+
+    dl = Libdl.dlopen_e("libccalltest")
+    @test dl != C_NULL
+
+    @test Libdl.dlclose(dl)
+    @test !Libdl.dlclose(dl)
```
Comment (staticfloat) [inline test/libdl.jl]: This line is failing on win32, according to Appveyor

**Comment by jdjohnston** [comment]
If it's line #197 that's failing, perhaps Win32 FreeLibrary doesn't error on calling it twice on the same handle. That only makes sense to me if the refcnt was > 1 before the first call. Without a Win32 Windows machine, this is going to be difficult to debug.

If indeed it is line #197 and unless someone can suggest a more reliable method to have FreeLibrary fail (other than a NULL handle, which is trapped by jl_dlclose), I would recommend deleting the line from the PR. :(

**Comment by StefanKarpinski** [comment]
> I would recommend deleting the line from the PR.

Or just change it to `@test_skip` so that there's a reminder to fix it if possible at some point in the future. Not high urgency, but it would be nice at some point. It's also a little unclear if we really need this return value or not.

**Comment by jdjohnston** [comment]
@staticfloat @StefanKarpinski , PR updated as recommended

**Review by StefanKarpinski — APPROVED** [review]

**Pull request merged**